test/GitcoinGovernor.t.sol

```diff
@@ -427,3 +428,466 @@ contract GitcoinGovernorAlphaPostProposalTest is GitcoinGovernorProposalTestHelp
+
+    // Ensure the proposal is queued
+    _state = governorBravo.state(_newProposalId);
+    assertEq(_state, IGovernor.ProposalState.Queued);
+
+    jumpPastProposalEta(_newProposalId);
+
+    // Execute the proposal
+    governorBravo.execute(_targets, _values, _calldatas, keccak256(bytes(_description)));
+
+    // Ensure the proposal is executed
+    _state = governorBravo.state(_newProposalId);
+    assertEq(_state, IGovernor.ProposalState.Executed);
+
+    // Confirm that governance settings have updated.
+    assertEq(governorBravo.votingDelay(), _newDelay);
+    assertEq(governorBravo.votingPeriod(), _newVotingPeriod);
+    assertEq(governorBravo.proposalThreshold(), _newProposalThreshold);
+  }
+
+  function testFuzz_NewGovernorCanPassMixedProposal(uint256 _amount, address _receiver) public {
+    IERC20 _token = _randomERC20Token(_amount);
+    assumeReceiver(_receiver);
+    uint256 _timelockTokenBalance = _token.balanceOf(TIMELOCK);
+
+    // bound by the number of tokens the timelock currently controls
+    _amount = bound(_amount, 0, _timelockTokenBalance);
+    uint256 _initialTokenBalance = _token.balanceOf(_receiver);
+
+    upgradeToBravoGovernor();
+    (
+      uint256 _newProposalId,
+      address[] memory _targets,
+      uint256[] memory _values,
+      bytes[] memory _calldatas,
+      string memory _description
+    ) = submitTokenSendProposal(address(_token), _amount, _receiver);
+
+    // Ensure proposal is in the expected state
+    IGovernor.ProposalState _state = governorBravo.state(_newProposalId);
+    assertEq(_state, IGovernor.ProposalState.Pending);
+
+    jumpToActiveProposal(_newProposalId);
+
+    // Ensure the proposal is now Active
+    _state = governorBravo.state(_newProposalId);
+    assertEq(_state, IGovernor.ProposalState.Active);
+
+    // Delegates vote with a mix of For/Against/Abstain with For winning.
+    vm.prank(PROPOSER); // kbw.eth (~1.8M votes)
+    governorBravo.castVote(_newProposalId, FOR);
+    vm.prank(0x2df9a188fBE231B0DC36D14AcEb65dEFbB049479); // janineleger.eth (~1.53M)
+    governorBravo.castVote(_newProposalId, FOR);
+    vm.prank(0x4Be88f63f919324210ea3A2cCAD4ff0734425F91); // kevinolsen.eth (~1.35M)
+    governorBravo.castVote(_newProposalId, FOR);
+    vm.prank(0x34aA3F359A9D614239015126635CE7732c18fDF3); // Austin Griffith (~1.05M)
+    governorBravo.castVote(_newProposalId, AGAINST);
+    vm.prank(0x7E052Ef7B4bB7E5A45F331128AFadB1E589deaF1); // Kris Is (~1.05M)
+    governorBravo.castVote(_newProposalId, ABSTAIN);
+    vm.prank(0x5e349eca2dc61aBCd9dD99Ce94d04136151a09Ee); // Linda Xie (~1.02M)
+    governorBravo.castVote(_newProposalId, AGAINST);
+
+    jumpToVotingComplete(_newProposalId);
+
+    // Ensure the proposal has succeeded
+    _state = governorBravo.state(_newProposalId);
+    assertEq(_state, IGovernor.ProposalState.Succeeded);
+
+    // Queue the proposal
+    governorBravo.queue(_targets, _values, _calldatas, keccak256(bytes(_description)));
```

Comment:
Other tests have the `// Ensure the proposal is queued` assertion after this, seems that's missing here? I know you hate spending time on test arch but there is a lot of duplicated logic in these tests that perhaps should be deduped to prevent missing things like this. Or maybe it's left out intentionally since it was already tested elsewhere? Which is also confusing then since it's not clear if it's intentional, which feels like another argument in favor of deduping hehe 😛 (since it can't hurt to test it twice if DRY'd)

Reply:
Yeah, I think it just didn't seem necessary to test that again and again, especially given that later assertions would fail if the proposal wasn't actually queued.

> there is a lot of duplicated logic in these tests that perhaps should be deduped to prevent missing things like this

You're not wrong! Let me think about this and try some things out. Generally I think explicitness and simplicity outweigh DRY-ness in tests. But it's a balance and sometimes it's worth it to add (minor) abstractions

Reply:
Cool, ultimately not a blocker so just leave a comment with whatever you decide and feel free to resolve this comment yourself

Reply:
Just took a look again. There is definitely repetition in these tests -- e.g. in the proposal state assertions, in the proposal lifecycle calls -- but there wasn't enough common ground for any clean abstraction to be obvious to me. Leaving as is for now, happy to revisit if/when we extend this test suite further